### PR TITLE
Adds RA.omitBy

### LINKED ISSUE #171: omitBy
Had use for `omitWhen` yesterday, open to a PR? https://github.com/ramda/ramda/wiki/Cookbook#remove-a-subset-of-keys-from-an-object-whose-associated-values-satisfy-a-given-predicate

Ref: #27

### PR DESCRIPTION
**Pull request by GingerPlusPlus**
Resolves #171.

Because I don't have experience writing tests, they're based on [Ramda's tests of `pickBy`](https://github.com/ramda/ramda/blob/v0.25.0/test/pickBy.js).

### CODE REVIEW

**Comment by guillaumearm** [comment]
Hi @GingerPlusPlus, thank you to help us growing `ramda-adjunct` 👍

Just few things to see : 

-----------------------

- Before all, please take a look to the [Commits Message Guidelines](https://github.com/char0n/ramda-adjunct/blob/master/CONTRIBUTING.md#-commit-message-guidelines) part in `CONTRIBUTING.md` and rename your commit message :

```bash
$ git commit --amend -m 'feat: add omitBy'
$ git push -f
```

This is because `CHANGELOG.md` file is auto-generated using [conventional-changelog-cli](https://github.com/conventional-changelog/conventional-changelog/tree/master/packages/conventional-changelog-cli)

-------------------------

- Secondly, you need to add signature types in `index.d.ts`

`ramda-adjunct` is fully support typescript, so you need to add typings for `RA.omitBy`

If you don't know typescript, we can help you with that (especially @BjornMelgaard).

-------------------------

Anyway,
implementation, documentation and test look good to me ✅

**Comment by GingerPlusPlus** [comment]
Ok, are type definitions in `index.d.ts` ordered somehow?

**Comment by BjornMelgaard** [comment]
@GingerPlusPlus they should be ordered as src/index.js

**Comment by guillaumearm** [comment]
I think we can have a more precise signature : 

```ts
omitBy<T, U extends Dictionary<T>>(
  pred: (val: T, key?: string, obj?: U) => boolean,
  obj: U,
): U;
```

@BjornMelgaard : what do you think ?

File: src/omitBy.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { complement, identity, pickBy, useWith } from 'ramda';
+
+/**
+ * Returns a partial copy of an object containing only the keys
+ * that don't satisfy the supplied predicate.
+ *
+ * @func omitBy
+ * @memberOf RA
```
Comment (char0n) [inline src/omitBy.js]: Insert  version pls
```js
* @since {@link https://char0n.github.io/ramda-adjunct/2.6.0|v2.6.0}
```

File: src/omitBy.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+ * @category Object
+ * @sig ((v, k) -> Boolean) -> {k: v} -> {k: v}
+ * @param {!Function} pred A predicate to determine whether or not a key
```
Comment (char0n) [inline src/omitBy.js]: Don't wrap lines. We're using eslint comments to disable long line issues

+ drop the `.` character at the end of the sentence

File: src/omitBy.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import { complement, identity, pickBy, useWith } from 'ramda';
+
+/**
```
Comment (char0n) [inline src/omitBy.js]: Insert before
```
/* eslint-disable max-len */

File: src/omitBy.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+ * @param {!Function} pred A predicate to determine whether or not a key
+ *        should be included on the output object.
+ * @param {!Object} obj The object to copy from
+ * @return {!Object} A new object only with properties that don't satisfy `pred`
+ *
+ * @example
+ *
+ * const isLowerCase = (val, key) => key.toLowerCase() === key;
+ * RA.omitBy(isLowerCase, {a: 1, b: 2, A: 3, B: 4}); //=> {A: 3, B: 4}
+ */
```
Comment (char0n) [inline src/omitBy.js]: Insert after
```js
/* eslint-enable max-len */

**Comment by GingerPlusPlus** [comment]
@guillaumearm Looking at [the definition of `R.pickBy`](https://github.com/ramda/ramda/blob/3dfce5ee44d287b8a6c0460cf57fb9d5a2f468dc/source/pickBy.js#L24-L32), it always returns `object`, not `U`.

And I'm not sure if `obj` should be typed as `U extends Dictionary<T>`.

File: test/omitBy.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    eq(RA.omitBy(R.always(1), obj), {});
+  });
+
+  it('when returning falsy, keeps the key', function () {
+    eq(RA.omitBy(R.F, obj), obj);
+    eq(RA.omitBy(R.always(0), obj), obj);
+    eq(RA.omitBy(R.always(null), obj), obj);
+  });
+
+  it('is called with (val,key,obj)', function () {
```
Comment (char0n) [inline test/omitBy.js]: is called with `(val,key,obj)` -> is called with `(val, key, obj)`

**Comment by guillaumearm** [comment]
@char0n : of course

@GingerPlusPlus : if you want to rebase on top of master, you can add the ramda-adjunct repository in your git remotes : 

```bash
$ git remote add ra https://github.com/char0n/ramda-adjunct.git
# 'ra' is the name of the remote, you can call it 'yolo' if you want
```

and then

```bash
# on your branch (git checkout omitBy)
$ git fetch ra
$ git rebase ra/master
```

Personally, I always run a `npm run lint && npm run test && npm run build && npm run docs` before push.

I remain available if you have any questions.

**Comment by guillaumearm** [comment]
@GingerPlusPlus : if you have already rebased, do it again, I just merge a PR.

File: src/omitBy.js (hunk 1 above)
Comment (GingerPlusPlus) [reply src/omitBy.js]: Done

**Comment by GingerPlusPlus** [comment]
@guillaumearm improved TypeScript type definition.

@char0n implemented all your suggestions, prettier done nothing to my code.

File: src/omitBy.js (hunk 1 above)
Comment (GingerPlusPlus) [reply src/omitBy.js]: Done.

File: src/omitBy.js (hunk 4 above)
Comment (GingerPlusPlus) [reply src/omitBy.js]: Done.

**Comment by char0n** [comment]
Nice job!

**Comment by guillaumearm** [comment]
> prettier does nothing to my code ?

@GingerPlusPlus : Are you sure of that ? can you re-check on new master ?

I just run prettier on master branch and there is updates for `test/omitBy.js`.

Anyway, I created an issue for that kind of case, please check out #379

**Comment by GingerPlusPlus** [comment]
> I just run prettier on master branch and there is updates for `test/omitBy.js`.

@guillaumearm `prettier .` didn't work, so yesterday I ended up only running it on `src/omitBy.js`, didn't think about running it on `test/omitBy.js` as well, sorry about that. It was my first time with `prettier`, probably not the last one tho, I like that it can automatically fix long lines, unlike `eslint --fix` that I use now for formatting.

I'm a fan of automating things like formatting and linting, so there's no way to forget.

File: src/index.d.ts [hunk 6]
```diff
@@ -29,6 +29,8 @@ declare namespace RamdaAdjunct {
 
     interface Dictionary<T> { [key: string]: T; }
 
+    type DictPred<T> = (value: T, key: string) => boolean;
```
Comment (GingerPlusPlus) [inline src/index.d.ts]: Just realized, it doesn't take into account the 3rd parameter, probably better would be something like
```typescript
type DictPred<T, U extends Dictionary<T>> = (value: T, key: string, dictionary: U) => boolean;
```

Comment (guillaumearm) [reply src/index.d.ts]: You right, and I think `key` and `dictionary` arguments should be optional.
Would you want to PR ?

@char0n : should we make a v2.6.1 patch ?

Comment (guillaumearm) [reply src/index.d.ts]: woops, sorry, 2.6.0 is not released yet 😂

**Comment by guillaumearm** [comment]
@GingerPlusPlus : I suspected it, but it's fine, it reveals problems and we can treat them : #379

**Pull request merged**